sklearn/neighbors/approximate.py

```diff
@@ -189,11 +189,11 @@ class LSHForest(BaseEstimator, KNeighborsMixin, RadiusNeighborsMixin):
 
       >>> X_train = [[5, 5, 2], [21, 5, 5], [1, 1, 1], [8, 9, 1], [6, 10, 2]]
       >>> X_test = [[9, 1, 6], [3, 1, 10], [7, 10, 3]]
-      >>> lshf = LSHForest()
+      >>> lshf = LSHForest(random_state=42)
       >>> lshf.fit(X_train)  # doctest: +NORMALIZE_WHITESPACE
       LSHForest(min_hash_match=4, n_candidates=50, n_estimators=10,
```

Comment:
> I would use n_candidates=10, n_estimators=10 (the default values).

That's what the doctest is currently using but @jnothman was saying that it doesn't really make sense given that the data in the doctest example is tiny:

```
>>> X_train = [[5, 5, 2], [21, 5, 5], [1, 1, 1], [8, 9, 1], [6, 10, 2]]
>>> X_test = [[9, 1, 6], [3, 1, 10], [7, 10, 3]]
```

Quoting him from #5324:

> Oh! Apparently this example is reporting the exact nearest neighbors due to very generous parameters (n_candidates, n_estimators are both high with respect to the tiny data). We should be reducing the parameters and fixing random_state.

Here is the full doctest (from master) in case it helps:

```
>>> from sklearn.neighbors import LSHForest

>>> X_train = [[5, 5, 2], [21, 5, 5], [1, 1, 1], [8, 9, 1], [6, 10, 2]]
>>> X_test = [[9, 1, 6], [3, 1, 10], [7, 10, 3]]
>>> lshf = LSHForest()
>>> lshf.fit(X_train)  # doctest: +NORMALIZE_WHITESPACE
LSHForest(min_hash_match=4, n_candidates=50, n_estimators=10,
          n_neighbors=5, radius=1.0, radius_cutoff_ratio=0.9,
          random_state=None)
>>> distances, indices = lshf.kneighbors(X_test, n_neighbors=2)
>>> distances                                        # doctest: +ELLIPSIS
array([[ 0.069...,  0.149...],
       [ 0.229...,  0.481...],
       [ 0.004...,  0.014...]])
>>> indices
array([[1, 2],
       [2, 0],
       [4, 0]])
```